src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -190,18 +191,87 @@ ExpressionSyntax initializer
 				immutable = false;
 			}
 
-			// TODO: narrow inspection with initializer
-			//       possibly a narrower type (using initializers type) and
-			//       possibly ImmutableTypeKind.Instances if its a new T( ... ).
+			var stuff = GetStuffToCheckForMember( type, typeSyntax, initializer );
 
-			if( !m_context.IsImmutable( type, ImmutableTypeKind.Total, typeSyntax.GetLocation(), out var diagnostic ) ) {
+			// null is a signal that there is nothing fruther that needs to be checked
+			if ( stuff == null ) {
+				return immutable;

```

Comment:
It is important that this isn't `return true;` because of the `readonly` check above.